src/python/tests_extended/test_dft_based.py

```diff
@@ -0,0 +1,311 @@
+    'ShortGrainDropper',
+    'RollingWin_Pivot_Integration',
+    'Laglead_Pivot_Integration',
+}
+
+INSTANCES = {
+    'DateTimeSplitter_Simple': Pipeline([
+        DateTimeSplitter(prefix='dt') << 'tokens1',
+        ColumnSelector(drop_columns=[
+                                    'dtHour12', 'dtDayOfWeek', 'dtDayOfQuarter',
+                                    'dtDayOfYear', 'dtWeekOfMonth', 'dtQuarterOfYear',
+                                    'dtHalfOfYear', 'dtWeekIso', 'dtYearIso', 'dtMonthLabel',
+                                    'dtAmPmLabel', 'dtDayOfWeekLabel', 'dtIsPaidTimeOff','dtHolidayName'
+        ])
+    ]),
+    'DateTimeSplitter_Complex' : Pipeline([
+        DateTimeSplitter(prefix='dt') << 'tokens1',
+        ColumnSelector(drop_columns=[
+                                    'dtHour12', 'dtDayOfWeek', 'dtDayOfQuarter',
+                                    'dtDayOfYear', 'dtWeekOfMonth', 'dtQuarterOfYear',
+                                    'dtHalfOfYear', 'dtWeekIso', 'dtYearIso', 'dtMonthLabel',
+                                    'dtAmPmLabel', 'dtDayOfWeekLabel', 'dtIsPaidTimeOff','dtHolidayName'
+        ])
+    ]),
+    'ToKey_SimpleFloat': ToKeyImputer(),
+    'ToKey_SimpleDouble': ToKeyImputer(),
+    'ToKey_SimpleString': ToKeyImputer(),
+    'ToKey_2col_Double': ToKeyImputer(),
+    'ToKey_2col_Double_String': ToKeyImputer(),
+    'ToString_Numbers': ToString(columns={'f0.out': 'f0',
+                                          'f1.out': 'f1',
+                                          'f2.out': 'f2'}),
+    'ToString_Other_Types': ToString(columns={'f0.out': 'f0',
+                                              'f1.out': 'f1'}),
+    'TimeSeriesImputer_1grain_2gap': TimeSeriesImputer(time_series_column='ts',
+                                                       filter_columns=['c'],
+                                                       grain_columns=['grain'],
+                                                       impute_mode='ForwardFill',
+                                                       filter_mode='Include'),
+    'TimeSeriesImputer_1grain_1gap_2filtercolumn': TimeSeriesImputer(time_series_column='ts',
+                                                        grain_columns=['grain'],
+                                                        filter_columns=['c3', 'c4'],
+                                                        impute_mode='ForwardFill',
+                                                        filter_mode='Include'),
+    'TimeSeriesImputer_2grain_nogap': TimeSeriesImputer(time_series_column='ts',
+                                                       filter_columns=['c'],
+                                                       grain_columns=['grain'],
+                                                       impute_mode='ForwardFill',
+                                                       filter_mode='Include'),
+    'ShortGrainDropper': ShortDrop(columns={'colA1': 'colA'},
```

Comment:
>ShortGrainDropper [](start = 5, length = 17)

Pls add tests for: Drop all rows

Reply:
ONNXRuntimeError is thrown if there's no remaining rows. Drop all rows has been added to bad input test.